src/wallet/use-cases/update/update.spec.ts

```diff
@@ -0,0 +1,143 @@
+import { UpdateWalletDTO } from '@/wallet/dto';
+import { IWalletRepository } from '@/wallet/interfaces';
+import { UpdateWalletUseCase } from '@/wallet/use-cases';
+import { FindBankByIdUseCase } from '@/bank/use-cases';
+import { IBankRepository } from '@/bank/interfaces';
+import { Bank } from '@/bank/infra/entities';
+import { Wallet } from '@/wallet/infra/entities';
+
+describe('Update Wallet Use Case', () => {
+  let sut: UpdateWalletUseCase;
+  let findBankByIdUseCase: FindBankByIdUseCase;
+  let walletRepository: IWalletRepository;
+  let findByIdMock: jest.SpyInstance;
+  let updateMock: jest.SpyInstance;
+  let findBankByIdMock: jest.SpyInstance;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        UpdateWalletUseCase,
+        FindBankByIdUseCase,
+        {
+          provide: IWalletRepository,
+          useFactory: () => ({
+            update: jest.fn(),
+            findById: jest.fn(),
+          }),
+        },
+        {
+          provide: IBankRepository,
+          useFactory: () => ({
+            findById: jest.fn(),
+          }),
+        },
+      ],
+    }).compile();
+
+    sut = module.get<UpdateWalletUseCase>(UpdateWalletUseCase);
+    findBankByIdUseCase = module.get<FindBankByIdUseCase>(FindBankByIdUseCase);
+    walletRepository = module.get<IWalletRepository>(IWalletRepository);
+
+    findByIdMock = jest.spyOn(walletRepository, 'findById');
+    updateMock = jest.spyOn(walletRepository, 'update');
+    findBankByIdMock = jest.spyOn(findBankByIdUseCase, 'execute');
+  });
+
+  beforeEach(async () => {
+    findBankByIdMock.mockResolvedValue(bank);
+    updateMock.mockResolvedValue(NotFoundException);
+  });
+
+  const validId = '4e8b5d94-6b16-4a42-b6d1-dc58b553d109';
+  const invalidId = 'invalid-id';
+
+  const updateData: UpdateWalletDTO = {
+    id: validId,
+    bank_id: validId,
+    account_type: AccountType.CHECKING_ACCOUNT,
+    description: 'Nova descrição',
+  };
+
+  const bank = {
+    id: validId,
+    name: 'anyBank',
+    logo_url: 'anyURL',
+  } as Bank;
+
+  it('should be able to update a wallet', async () => {
+    const updatedWallet = updateData as unknown as Wallet;
+
+    findByIdMock.mockResolvedValue(updatedWallet);
+    updateMock.mockResolvedValue(updatedWallet);
+
+    const result = await sut.execute(updateData);
+
+    expect(result).toEqual(updatedWallet);
+
+    expect(findBankByIdUseCase.execute).toHaveBeenCalledTimes(1);
+    expect(findBankByIdUseCase.execute).toHaveBeenCalledWith(
+      updateData.bank_id,
+    );
+
+    expect(walletRepository.findById).toHaveBeenCalledTimes(1);
+    expect(walletRepository.findById).toHaveBeenCalledWith(updateData.id);
+
+    expect(walletRepository.update).toHaveBeenCalledTimes(1);
+    expect(walletRepository.update).toHaveBeenCalledWith(updatedWallet);
+  });
+
+  it('should not be able to update a wallet when wallet does not exists', async () => {
```

Comment:
"should not be able to update a wallet if wallet does not exist."

Tira o "s" do exist, e eu acho melhor manter a frase assim, pra manter o padrão dos nomes, como está no wallet.controller.e2e.spec.ts